integration/common.sh

```diff
@@ -29,6 +29,20 @@ waitForServerStartup() {
   sleep 2
 }
 
+pickUnusedPort() {
+  local base=$1
+  local port
+  while :; do
```

Comment:
nit: I think only the inner loop should be enough. Let it die if we can't pick a port after a certain number of iterations.